orttraining/orttraining/core/optimizer/compute_optimizer/padding_elimination.cc

```diff
@@ -175,108 +148,50 @@ NodeArg* InsertNodesForInput(Graph& graph,
+// Insert PadAndUnflatten to unflatten the shape of the in_index-th input of node.
 // The gathergrad_index_arg is the indices of the elements that are not padding.
 // The new_shape_arg is the shape of [batch_size * seqlen, ...]
 // gathergrad_index_arg and new_shape_arg are the arguments needed by GatherGrad.
 NodeArg* InsertNodesForOutput(Graph& graph,
                               Node& node,
                               uint32_t in_index,
                               NodeArg* gathergrad_index_arg,
-                              NodeArg* new_shape_arg,
                               NodeArg* first_two_dims_arg,
                               const logging::Logger& logger) {
-  std::vector<int64_t> other_indices;
-  auto input_shape = node.InputDefs()[in_index]->Shape();
-  for (int k = 2; k < input_shape->dim_size(); k++) {
-    // When executing, Shape of node here has been flattened, so the indices should be k-1.
-    other_indices.push_back(int64_t(k) - 1);
-  }
-
-  // Construct the unflattened_shape_arg of [batch_size, seqlen, ...]
-  NodeArg* unflattened_shape_arg = nullptr;
-  if (other_indices.empty()) {
-    unflattened_shape_arg = first_two_dims_arg;
-  } else {
-    // If the shape size of the in_index-th input of node is larger than 2 dims, we need to concat the first two dims
-    // of [batch_size, seqlen] and the other dims together.
-    ONNX_NAMESPACE::TensorProto other_indices_const_tensor;
-    other_indices_const_tensor.set_name(graph.GenerateNodeArgName("other_shape"));
-    other_indices_const_tensor.set_data_type(ONNX_NAMESPACE::TensorProto_DataType_INT64);
-    other_indices_const_tensor.add_dims(other_indices.size());
-    other_indices_const_tensor.set_raw_data(other_indices.data(), other_indices.size() * sizeof(int64_t));
-    NodeArg* other_indices_arg = &graph_utils::AddInitializer(graph, other_indices_const_tensor);
-    NodeArg* other_dims_arg = GetDimsValue(graph, node.MutableInputDefs()[in_index], other_indices_arg, node);
-
-    InlinedVector<NodeArg*> concat_input_args;
-    concat_input_args.push_back(first_two_dims_arg);
-    concat_input_args.push_back(other_dims_arg);
-
-    InlinedVector<NodeArg*> concat_output_args{&graph.GetOrCreateNodeArg(graph.GenerateNodeArgName("concat_shape_result"),
-                                                                         nullptr)};
-
-    onnxruntime::NodeAttributes attributes;
-    attributes["axis"] = ONNX_NAMESPACE::MakeAttribute("axis", int64_t(0));
-
-    Node& concat_node = graph.AddNode(graph.GenerateNodeName("concat_shape"), "Concat", "", concat_input_args,
-                                      concat_output_args, &attributes, kOnnxDomain);
-    ORT_ENFORCE(graph.SetOpSchemaFromRegistryForNode(concat_node), "Failed to concat shape for " + concat_node.Name());
-    concat_node.SetExecutionProviderType(node.GetExecutionProviderType());
-    unflattened_shape_arg = concat_output_args[0];
-  }
-
   InlinedVector<NodeArg*> gathergrad_input_args;
```

Comment:
rename the 'gathergrad_input_args' to 'pad_and_unflatten_args' ?

Reply:
updated